### ISSUE TITLE
Inconsistency between PyCOCOCallback and BoxCOCOMetrics

### ISSUE THREAD

**Issue by tu6berk** [keras-team/keras-cv#2402]
### Current Behavior:
While constructing the ground truth dataset, PyCOCOCallback computes the number of detections within annotations as follows:

`   num_detections = ops.sum(ops.cast(gt_classes > 0, "int32"), axis=-1)`
 
at [line 102](https://github.com/keras-team/keras-cv/blob/301f7cd565eca7b3ac32610a7ebeac6b0af5403f/keras_cv/callbacks/pycoco_callback.py#L102C9-L102C77).

In constrast, BoxCOCOMetrics tackles it via this approach:

`    ground_truth["num_detections"] = [
        ops.sum(ops.cast(y_true["classes"] >= 0, "int32"), axis=-1)
    ]`

at [lines 300-302](https://github.com/keras-team/keras-cv/blob/301f7cd565eca7b3ac32610a7ebeac6b0af5403f/keras_cv/metrics/object_detection/box_coco_metrics.py#L300C1-L302C6).

Consequently, the results produced by the two are quite different, especially when your data set has only a few distinct labels and the encoding is plain integer assignment with enumeration starting from 0.

### Expected Behavior:
Since label encoding schemes in most Keras CV models enumerate the classes starting from 0, which holds in my case as well, I believe that BoxCOCOMetrics approach should be applied to PyCOCOCallback. Alternatively this could be made a parameter to be determined by the user.

### Version:
- 0.8.2

**Comment by LukeWood** [comment]
I also agree that the `PyCOCOCallback` is incorrect here, and that it should be updated to match the `BoxCOCOMetrics` (Assuming that the package still uses 0 as the first class index across the repo)

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new **KerasHub** package, which supports image, text, and multi-modal models. Please read [the announcement](https://github.com/keras-team/keras-hub/issues/1831). `KerasHub` will support all the core functionality of `KerasCV`.

`KerasHub` can be installed with `!pip install -U keras-hub`. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
